config.go

```diff
@@ -153,6 +154,18 @@ func (fm *Frontman) userAgent() string {
 	return fmt.Sprintf("Frontman v%s %s %s", parts[0], runtime.GOOS, runtime.GOARCH)
 }
 
+func (fm *Frontman) DumpConfigToml() string {
+	buff := &bytes.Buffer{}
+	enc := toml.NewEncoder(buff)
+	err := enc.Encode(fm)
+
+	if err != nil {
+		log.Errorf("DumpConfigToml error: %s", err.Error())
```

Comment:
The function should return in error case

Reply:
No, it shouldn't. 
I'm using the non-nil `bytes.Buffer` to collect the encoded toml config.
If somehow TOML encoder returns an error, it doesn't mean that it doesn't already push some useful data into the buffer. 

If it doesn't push anything to the buffer, `buff.String()` will return empty string

Reply:
I don't think a partial output is good in case there occurred an error. Output could be wrong or invalid.

The fact that `buff.String()` returns an empty string is not explicit but just because there are no more instructions between error case handling and the return statement. This is obvious now but if someone else edit the function the next time this can be overseen and introduce a bug or unexpected behaviour. So please don't "couple" the error handling code with code unrelated to the error. _All_ the error handling should be done within the `if err` branch and not rely on code after the branch.

Reply:
ok, let it be like this. BTW, we have the same code in the cagent